pipeline/metadata/schema.py

```diff
@@ -93,6 +53,49 @@ def merge_ip_metadata(base: IpMetadata, new: IpMetadataWithKeys) -> None:
     base.name = new.name
 
 
+@dataclass
+class SatelliteTags():
+  """Satellite tags that are not metadata specifically about the answer ip"""
+  http: Optional[str] = None
+  cert: Optional[str] = None
+  matches_control: Optional[str] = None
+
+
+@dataclass
+class SatelliteAnswer():
+  """Class for keeping track of Satellite answer content"""
+  # Keys
+  ip: str
+  tags: SatelliteTags = dataclasses.field(default_factory=SatelliteTags)
+  ip_metadata: IpMetadata = dataclasses.field(default_factory=IpMetadata)
+
+
+@dataclass
+class SatelliteAnswerWithKeys(SatelliteAnswer):
+  """Satellite Answer Metadata.
+
+  When this metadata is being passed around
+  it needs an additional date field to keep track of when it's valid.
+  """
+  date: str = ''
+
+
+def merge_satellite_tags(base: SatelliteTags, new: SatelliteTags) -> None:
+  if new.http is not None:
+    base.http = new.http
+  if new.cert is not None:
+    base.cert = new.cert
+  if new.matches_control is not None:
+    base.matches_control = new.matches_control
+
+
+def merge_satellite_answers(base: SatelliteAnswer,
+                            new: SatelliteAnswer) -> None:
+  """Add tag information to a Satellite answer"""
+  merge_satellite_tags(base.tags, new.tags)
+  merge_ip_metadata(base.ip_metadata, new.ip_metadata)
+
+
 @dataclass
 class ReceivedHttps:
```

Comment:
Optional: This is a weird name. Maybe `HttpsResponse` would be clearer.

Reply:
done